sway-core/src/type_system/type_engine.rs

```diff
@@ -44,6 +44,17 @@ impl TypeEngine {
         }
     }
 
+    /// Currently the [TypeEngine] is a lazy static object, so when we run
+    /// cargo tests, we can either choose to use a local [TypeEngine] and bypass
+    /// all of the global methods or we can use the lazy static [TypeEngine].
+    /// This method is for testing to be able to bypass the global methods for
+    /// the lazy static [TypeEngine] (contained within the call to hash in the
+    /// id_map).
+    #[allow(dead_code)]
```

Comment:
If this is only for testing, maybe we can add a `#[cfg(test)]`?
```suggestion
    #[cfg(test)]
    #[allow(dead_code)]
```